FWCore/Framework/src/ProductResolvers.h

```diff
@@ -422,6 +423,12 @@ namespace edm {
                                bool skipCurrentProcess,
                                SharedResourcesAcquirer* sra,
                                ModuleCallingContext const* mcc) const override {
+      if (principal.branchType() == InProcess &&
+          (mcc->parent().globalContext()->transition() == GlobalContext::Transition::kBeginProcessBlock ||
+           mcc->parent().globalContext()->transition() == GlobalContext::Transition::kEndProcessBlock)) {
+        return Resolution(nullptr);
+      }
```

Comment:
Just to check I understood the code correctly, does this mean that ProcessBlock products from parent processes can not be accessed in begin/endProcessBlock transitions by a SubProcess? (makes sense, now that I wrote it down)

Reply:
Yes.

I was struggling with SubProcesses in this PR for a while.

This is the one and only place where the behavior that was in the transient ProcessBlock PR last year changed. If processes are run as separate cmsRun processes, then one can access a ProcessBlock from a previous process only in the AccessInputProcessBlock transition. In the current PR, the same behavior occurs when those processes are run in one cmsRun process with SubProcesses. ProcessBlock products from a parent process are only accessible to a SubProcess in the AccessInputProcessBlock transition.

That AccessInputProcessBlock transition occurs and the ProcessBlock products from parent processes are available in a SubProcess. The issue that occurs is that this transition does not occur until after all the Event transitions are complete so the caches will not be filled during the Event transitions (cacheHandles will be invalid).

Reply:
One other related comment. Normally the AccessInputProcessBlock transitions run immediately after opening an input file. In a SubProcess, when the transition is related to a parent process of the SubProcess (not related to products in the input file) the AccessInputProcessBlock transition will run near the end, just after the parent process's endProcessBlock transition.

Reply:
Thanks. I think it makes sense (as I wrote earlier). I would imagine this behavior to limit the usefulness of passing ProcessBlock products from parent process to SubProcess, but I can't think of any way around it, since the behavior follows rather naturally from the principles of the SubProcess itself.